python/docs/lis/userguide.rst

```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
+    >>> format_spec.direction
+    255
+
+This tells us that the current logset is indexed against depth, and that the
+depth is measured in 1/10 of an inch. There is also a constant spacing of 60
+.1IN (or 6 IN) between each sample. Note that there is no general requirement
+that the index is evenly spaced and these values might be missing in such
+cases. Lastly we see that the measurements is taken going down-hole (refer to
+:attr:`dlisio.lis.DataFormatSpec.direction`).
+
+Now lets look at the individual channels/curves in the logset. We can list the
+mnemonics and the units of measurement by simply looping the Spec Blocks:
+
+.. code-block:: python
+
+    >>> for spec in format_spec.specs:
+    ...     print("Index: {} (units={})".format(spec.mnem, spec.units))
+    "Index: 'TDEP', units: '.1IN')"
+    "Index: 'TIME', units: 'ms  ')"
+    "Index: 'CHX ', units: 'M   ')"
+    "Index: 'RCCL', units: '    ')"
 
 .. note::
-   LIS79 opens for the presence of duplicated DFSR, for redundancy. Currently,
-   dlisio has no support for identifying redundant DFSRs, and curves() will
-   return empty arrays for these.
+    Please refer to :class:`dlisio.lis.DataFormatSpec` and
+    :class:`dlisio.core.spec_block_0` / :class:`dlisio.core.spec_block_1` for
+    the complete documentation of DFSR and Spec Blocks, respectively.
 
+Reading the curves
+..................
+
+The :class:`dlisio.lis.DataFormatSpec` only contains the metadata about the
+logset. To read the actual curves, pass the DFSR ``format_spec`` to
+:func:`dlisio.lis.curves` together with the filehandle ``f``:
+
+.. code-block:: python
+
+    >>> curves = lis.curves(f, format_spec)
+
+This returns a `structured numpy.ndarray`_ containing all the curves in the
+logset, using the mnemonics from the Spec Blocks as field (column) names.
+
+Fast Channels
+.............
+
+There is one caveat to reading the curves from a logset that is necessary to be
+aware of. LIS supports a concept it refers to as 'Fast Channels'. In short this
+means that channels within the *same* logset can have different sampling rates.
+
+When a logset contains curves with different sampling rates, these have to be
+read separately. I.e. multiple calls to the :func:`dlisio.lis.curves` is
+necessary in order to read all the curves. The sampling rate of a channel is
+recorded in its Spec Block:
+
+.. code-block:: python
+
+    >>> ch = format_spec.specs[-1]
+    >>> ch.mnem
+    'RCCL'
+    >>> ch.samples
+    6
+
+The last channel in this Data Format Spec, ``RCCL``, has a sampling rate of 6,
+which means it's sampled 6 times as frequent as the index channel. Remember,
+that the index ``DEPT`` was sampled once every 6th inch. That implicitly means
+that ``RCCL`` is sampled once every inch.
+
+You can also get a set of all sample rates in the DataFormatSpec:
+
+.. code-block:: python
+
+    >>> format_spec.sample_rates()
+    {1, 6, 30}
+
+There are 3 different sampling rates, which be must read with separate calls to
```

Comment:
which (must be) read

And maybe not read, but "which must be used in separate calls to ...". After all, you are are not reading the rates, you read the data.